src/Compilers/CSharp/Portable/Symbols/Source/CustomModifierUtils.cs

```diff
@@ -63,6 +63,14 @@ internal static TypeSymbol CopyTypeCustomModifiers(TypeSymbol sourceType, TypeSy
 
             if (sourceType.IsTupleType)
             {
+                // TODO(https://github.com/dotnet/roslyn/issues/12389):
+                // Need to save/restore tupleness as well
+                if (sourceType.IsTupleType)
```

Comment:
Note: we may want to keep a check like this for the future since it's really unlikely to have any custom modifiers and this can produce a moderate amount of allocations even if there's no real work to be done.